modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/layer/SymbolMBLayer.java

```diff
@@ -1731,7 +1731,12 @@ public Expression textTranslateAnchor() {
         // layer.textHaloBlur();
 
         // auto wrap
-        // symbolizer.getOptions().put("autoWrap", layer.textMaxWidth()); //  Pixels (GS) vs ems (MB); Vendor options with expressions?
+        // uses getTextSize to convert text-max-width from ems to pixels
+        if(getTextMaxWidth() != null) {
+            symbolizer.getOptions().put("autoWrap", String.valueOf(getTextMaxWidth().intValue() * getTextSize().intValue())); 
+        } else{
+            symbolizer.getOptions().put("autoWrap", String.valueOf(getTextSize().intValue() * 10));
```

Comment:
10 is the default value for ems in the Mapbox docs, I'll look into the StyleFactory and see what I can find, and work on a test case thanks!